orttraining/orttraining/training_api/include/onnxruntime_training_cxx_inline.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include "onnxruntime_training_c_api.h"
+#include "onnxruntime_cxx_api.h"
+
+namespace Ort {
+
+inline TrainingSession::TrainingSession(const SessionOptions& session_options,
+                                        CheckpointState& checkpoint_state,
+                                        const std::basic_string<ORTCHAR_T>& train_model_path,
+                                        const std::optional<std::basic_string<ORTCHAR_T>>& eval_model_path,
+                                        const std::optional<std::basic_string<ORTCHAR_T>>& optimizer_model_path) {
+  Env env = Env();
+  ThrowOnError(GetTrainingApi().CreateTrainingSession(
+      env, session_options, checkpoint_state,
+      train_model_path.c_str(),
+      eval_model_path.has_value() ? eval_model_path.value().c_str() : nullptr,
+      optimizer_model_path.has_value() ? optimizer_model_path.value().c_str() : nullptr,
+      &p_));
+
+  ThrowOnError(GetTrainingApi().TrainingSessionGetTrainingModelOutputCount(p_, &training_model_output_count_));
+
+  ThrowOnError(GetTrainingApi().TrainingSessionGetEvalModelOutputCount(p_, &eval_model_output_count_));
+}
+
+inline std::vector<Value> TrainingSession::TrainStep(const std::vector<Value>& input_values) {
+  std::vector<Value> output_values;
+  for (size_t i = 0; i < training_model_output_count_; i++) output_values.emplace_back(nullptr);
+  auto ort_input_values = reinterpret_cast<const OrtValue**>(const_cast<Value*>(input_values.data()));
+  auto ort_output_values = reinterpret_cast<OrtValue**>(output_values.data());
+  RunOptions run_options;
```

Comment:
Make RunOptions a function parameter? This will allow the users to pass in RunOptions. API for creating RunOptions already exists, plus C# bindings allow passing in RunOptions too.

Reply:
same comment is applicable for Eval and Optimizer steps too

Reply:
The reason why I did not expose `RunOptions` for now is because as of now, they get [dropped anyway](https://github.com/microsoft/onnxruntime/blob/c20abcab87360eeea5de72fb951d5466adfacc99/orttraining/orttraining/training_api/training_session.cc#L54-L64).

I think we can add another overloaded method as needed?